volatility3/framework/plugins/linux/hidden_modules.py

```diff
@@ -29,46 +30,30 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             requirements.PluginRequirement(
                 name="lsmod", plugin=lsmod.Lsmod, version=(2, 0, 0)
             ),
+            requirements.VersionRequirement(
+                name="linux_utilities_modules",
+                component=linux_utilities_modules.Modules,
+                version=(2, 0, 0),
+            ),
         ]
 
-    @classmethod
+    @staticmethod
+    @Deprecation.deprecated_method(
+        replacement=linux_utilities_modules.Modules.get_modules_memory_boundaries,
+        replacement_version=(2, 0, 0),
+    )
     def get_modules_memory_boundaries(
-        cls,
         context: interfaces.context.ContextInterface,
         vmlinux_module_name: str,
-    ) -> Tuple[int]:
-        """Determine the boundaries of the module allocation area
-
-        Args:
-            context: The context to retrieve required elements (layers, symbol tables) from
-            vmlinux_module_name: The name of the kernel module on which to operate
-
-        Returns:
-            A tuple containing the minimum and maximum addresses for the module allocation area.
-        """
-        vmlinux = context.modules[vmlinux_module_name]
-        if vmlinux.has_symbol("mod_tree"):
-            # Kernel >= 5.19    58d208de3e8d87dbe196caf0b57cc58c7a3836ca
-            mod_tree = vmlinux.object_from_symbol("mod_tree")
-            modules_addr_min = mod_tree.addr_min
-            modules_addr_max = mod_tree.addr_max
-        elif vmlinux.has_symbol("module_addr_min"):
-            # 2.6.27 <= kernel < 5.19   3a642e99babe0617febb6f402e1e063479f489db
-            modules_addr_min = vmlinux.object_from_symbol("module_addr_min")
-            modules_addr_max = vmlinux.object_from_symbol("module_addr_max")
-
-            if isinstance(modules_addr_min, objects.Void):
-                raise exceptions.VolatilityException(
-                    "Your ISF symbols lack type information. You may need to update the"
-                    "ISF using the latest version of dwarf2json"
-                )
-        else:
-            raise exceptions.VolatilityException(
-                "Cannot find the module memory allocation area. Unsupported kernel"
-            )
-
-        return modules_addr_min, modules_addr_max
+    ) -> Tuple[int, int]:
+        return linux_utilities_modules.Modules.get_modules_memory_boundaries(
+            context, vmlinux_module_name
+        )
 
+    @Deprecation.deprecated_method(
+        replacement=linux_utilities_modules.Modules.get_module_address_alignment,
+        replacement_version=(2, 0, 0),
+    )
     @classmethod
     def _get_module_address_alignment(
```

Comment:
Surprised I let a private method be exposed as a classmethod, but just as well to deprecate it, thanks!

Reply:
(It probably had to be a classmethod to be called from within another classmethod, that makes sense).